src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1810 @@
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+
+HIFIGAN_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`FastSpeech2ConformerConfig`]):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+FASTSPEECH2_CONFORMER_WITH_HIFIGAN_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`FastSpeech2ConformerWithHifiGanConfig`]):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+
+class FastSpeech2ConformerDurationPredictorLayer(nn.Module):
+    def __init__(self, config: FastSpeech2ConformerConfig, layer_idx=0):
+        super().__init__()
+        num_chans = config.duration_predictor_channels
+        kernel_size = config.duration_predictor_kernel_size
+        input_channels = config.hidden_size if layer_idx == 0 else num_chans
+        self.layer = nn.ModuleList(
+            [
+                nn.Conv1d(
+                    input_channels,
+                    num_chans,
+                    kernel_size,
+                    stride=1,
+                    padding=(kernel_size - 1) // 2,
+                ),
+                nn.ReLU(),
+                FastSpeech2ConformerDimensionalLayerNorm(num_chans),
+                nn.Dropout(config.duration_predictor_dropout_rate),
+            ]
+        )
```

Comment:
Why use ModuleList when you can just define `self.conv, self.act_fn, self.norm, self.dropout`?